RNWCPP/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +30,289 @@ enum class ScrollViewCommands
   ScrollToEnd,
 };
 
+class ScrollViewShadowNode : public ShadowNodeBase
+{
+  using Super = ShadowNodeBase;
+public:
+  ScrollViewShadowNode();
+  void dispatchCommand(int64_t commandId, const folly::dynamic& commandArgs) override;
+  void createView() override;
+  void updateProperties(const folly::dynamic&& props) override;
+
+  // snaps to the snapToInterval based on the snapToAlignment (start, center, end) within the visable area as defined by the viewableSize
+  int snap(double original, winrt::Primitives::SnapPointsAlignment snapToAlignment, double snapToInterval, double viewableSize);
+
+protected:
+  double m_snapToInterval;
+  bool m_isSnapToAlignmentSet;
+  winrt::Primitives::SnapPointsAlignment m_snapToAlignment;
+
+private:
+   void topScroll(winrt::ScrollViewer& scrollViewer, int64_t tag, const char* event, double x, double y, double zoom);
+};
+
+ScrollViewShadowNode::ScrollViewShadowNode()
+{
+  m_snapToInterval = 0;
+  m_isSnapToAlignmentSet = false;
+  m_snapToAlignment = winrt::Primitives::SnapPointsAlignment::Near;
```

Comment:
I prefer initializing members where they're declared (a C++ 11 feature).  I'd like to see that convention adopted in this code.